src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.Tables.cs

```diff
@@ -45,33 +46,132 @@ public virtual List<PSTable> FilterPSTables(string resourceGroupName, string wor
+        {
+            PSTable existingTable = this.GetTable(resourceGroupName, workspaceName, tableName);
+
+            return OperationalInsightsManagementClient.Tables.MigrateWithHttpMessagesAsync(resourceGroupName, workspaceName, tableName).Result.Response.StatusCode;
+        }
+
+        public virtual HttpStatusCode DeletePSTable(string resourceGroupName, string workspaceName, string tableName)
+        {
+            PSTable existingTable = this.GetTable(resourceGroupName, workspaceName, tableName);
+
+            var res = OperationalInsightsManagementClient.Tables.DeleteWithHttpMessagesAsync(resourceGroupName, workspaceName, tableName).Result;
+
+            return res.Response.StatusCode;
+        }
+
+        public virtual PSTable CreateRestoreTable(PSRestoreTable properties)
+        {
+            ValidateTableNotExist(properties, TableConsts.RestoredLogsSuffix);
+            var response = OperationalInsightsManagementClient.Tables.CreateOrUpdate(properties.ResourceGroupName, properties.WorkspaceName, properties.TableName, properties.ToTableProperties());
+
+            return new PSTable(response, properties.ResourceGroupName, properties.WorkspaceName);
+        }
+
+        public virtual PSTable CreateSearchTable(PSSearchTable properties)
+        {
+            ValidateTableNotExist(properties, TableConsts.SearchResultsSuffix);
+            var response = OperationalInsightsManagementClient.Tables.CreateOrUpdate(properties.ResourceGroupName, properties.WorkspaceName, properties.TableName, properties.ToTableProperties());
+
+            return new PSTable(response, properties.ResourceGroupName, properties.WorkspaceName);
+        }
+
+        public virtual PSTable CreatePSTable(PSTable properties)
+        {
+            ValidateTableNotExist(properties, TableConsts.CustomLogSuffix);
+            var response = OperationalInsightsManagementClient.Tables.CreateOrUpdate(properties.ResourceGroupName, properties.WorkspaceName, properties.TableName, properties.ToTableProperties());
+
+            return new PSTable(response, properties.ResourceGroupName, properties.WorkspaceName);
+        }
+
+        private void ValidateTableNotExist(PSTable properties, string suffix)
+        {
+            PSTable existingTable = null;
+            try
+            {
+                existingTable = this.GetTable(properties.ResourceGroupName, properties.WorkspaceName, properties.TableName);
+            }
+            catch (PSArgumentException)
+            {
+                // Do nothing as GetTable should throw an exception
+            }
+
+            if (existingTable != null)
+            {
+                throw new PSArgumentException(string.Format(Constants.TableAlreadyExist, properties.TableName, properties.ResourceGroupName, properties.WorkspaceName));
+            }
+
+            if (!properties.TableName.Substring(properties.TableName.Length - suffix.Length).ToUpperInvariant().Equals(suffix))
```

Comment:
properties.TableName.Substring(properties.TableName.Length - suffix.Length)
this line get the table name without suffix (if there are any), do you mean to compare this with the suffix?